cas-server-core/src/main/java/org/jasig/cas/util/JCEBasedCipherExecutor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.util;
+
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.keys.AesKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import javax.validation.constraints.NotNull;
+
+/**
+ * A {@link CipherExecutor} implementation that is based on algorithms
+ * provided by the default platform's JCE. By default AES encryption is
+ * used which requires both the secret key and the IV length to be of size 16.
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+public class JCEBasedCipherExecutor implements CipherExecutor<byte[], byte[]> {
+    private static final String UTF8_ENCODING = "UTF-8";
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /** Cipher algorithm. Default is <code>AES/CBC/NoPadding</code>. */
+    private String cipherAlgorithm = "AES/CBC/PKCS5Padding";
```

Comment:
I'm always surprised to see all this plumbery as we have Spring everywhere which has a good hashing / encryption support or as we could use shiro-core which also has a strong hashing / encryption support.